File: BackEnd/AnayaCore/Models/Product.cs
namespace AnayaCore.Models
{
    public class Product
    {
        public int Id { get; set; } = 0;
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public int? CategoryId { get; set; } = 0;
        public int Price { get; set; } = 0;
        public string ImageUrl { get; set; } = string.Empty;
        public int Quantity { get; set; } = 1;
    }
}

File: FrontEnd/AnayaUI/src/app/shared/models/product.ts
/**
 * Product Model
 * Represents a product in the e-commerce system
 */
export interface Product {
  id: number;
  name: string;
  description: string;
  categoryId: number | null;
  price: number;
  imageUrl: string;
  quantity?: number;
}

/**
 * Product Filter Options
 */
export interface ProductFilter {
  searchTerm?: string;
  categoryId?: number | null;
  minPrice?: number;
  maxPrice?: number;
}
